overlord/state/change.go

```diff
@@ -630,3 +631,75 @@ func (c *Change) abortTasks(tasks []*Task, abortedLanes map[int]bool, seenTasks
 		c.abortLanes(lanes, abortedLanes, seenTasks)
 	}
 }
+
+// VerifyTaskDependencies checks the tasks in the change for cyclic dependencies
+// and returns in error in such case.
+func (c *Change) VerifyTaskDependencies() error {
+	tasks := c.Tasks()
+	// list of successors of a task (their IDs)
+	successors := make(map[string][]string, len(tasks))
```

Comment:
but do we need this?  we don't mutate afaict atm, isn't it task.HaltTasks() ?

Reply:
yup, dropped the list now